AsyncDisplayKit/ASTableView.mm

```diff
@@ -789,4 +842,23 @@ - (NSUInteger)dataControllerNumberOfSections:(ASDataController *)dataController
   }
 }
 
+#pragma mark - _ASTableViewCellDelegate
+
+- (void)tableViewCell:(UITableViewCell *)cell atIndexPath:(NSIndexPath *)indexPath didTransitionToState:(UITableViewCellStateMask)state
+{
+  [self beginUpdates];
+  ASCellNode *node = [_dataController nodeAtIndexPath:indexPath];
+  
+  CGSize constrainedSize = [self dataController:_dataController constrainedSizeForNodeAtIndexPath:indexPath];
+  if (state != UITableViewCellStateDefaultMask) {
+    // Edit control or delete confirmation was shown and size of content view was changed.
+    // The new size should be taken into consideration.
+    constrainedSize.width = MIN(cell.contentView.frame.size.width, constrainedSize.width);
+  }
+  
+  [node measure:constrainedSize];
+  node.frame = CGRectMake(0, 0, node.calculatedSize.width, node.calculatedSize.height);
```

Comment:
In an ideal world, this might be done in the UITableViewCell subclass itself — even in the -layoutSubviews method, perhaps.  Not something to worry about now, maybe it would break something